app/src/ui/changes/commit-message.tsx

```diff
@@ -235,6 +237,9 @@ export class CommitMessage extends React.Component<
       isCommittingStatusMessage: '',
       repoRulesEnabled: false,
       isRuleFailurePopoverOpen: false,
+      // TODO: do a proper retrieval rather than pinging localStorage (use AppState?)
+      // (current solution requires a tab switch to hot reload)
+      showCommitLengthWarning: localStorage.getItem("showCommitLengthWarning") === "1",
```

Comment:
The `getBoolean` you are using on line 556 is the call to the localStorage so you were already on the right track. :) When you put that in the `loadInitialState`, it will populate the appState correctly from local storage. Then, you need to pass it down through the props, I believe if you follow `commitSpellcheckEnabled`, you can see how to it gets from `App.tsx` in the `renderRepository` down through multiple components to this file. :) 

Then you will no longer needs a `showCommitLengthWarning` state variable on this component as you will be able to use `this.props.showCommitLengthWarning`